### ISSUE TITLE
Is this project alive?

### ISSUE THREAD

**Issue by patrickalima98** [denorg/scrypt#12]
Hello, I noticed that this library hasn't received any updates for over a year, I would like to know if the project is still active?

**Comment by AnandChowdhary** [comment]
Only if @oplik0 is still available if you need help, but I'm unable to maintain this for the foreseeable future.

**Comment by oplik0** [comment]
I am, there simply isn't really much that needs to change.
The only "bigger" thing I'll probably do soon is adjust default parameters to current OWASP recommendations (since the current N value is a bit lower).
Other than that, this doesn't even really need dependency bumps that much - it's only using a few std dependencies that didn't change in functionality (though I'll probably bump them for cache reasons) and even the rust scrypt crate only really got the default parameter update and some API improvements.

**Comment by oplik0** [comment]
Released 4.3.0 - as mentioned, the changes aren't big.

I traded a small bit of performance (few % difference) for losing over 20KB in size (actually I think the perf loss was at least in part a consequence of updating dependecies and not my config changes, but I didn't feel like benchmarking everything), bumped default logN to 17, and bumped std dependency.

Though because synchronous hmac was removed from Deno STD in favor of asynchronous WebCrypto implementation I quickly reimplemented one using `crypto.digestSync` - it's only used for the weird original scrypt output encoding (where the hash is the key used for hashing the parameters...), and with the data used there it seems to preform comparably to old std one (in general it seems faster but with higher constant costs, so for larger inputs it leads and for smaller ones it loses on perf. I probably could try to optimize it more, but again - it's neither performance nor even security critical here).

So tl;dr - dependency bumps, default params update, tiny performance regression in exchange for more significant decrease in package size, functionality and API unchanged.

So, unless some issue comes up I probably won't be doing much here - just bumping the standard library dependency (I guess an unmentioned change was moving this to an import map so now it's a one line update) from time to time.

**Comment by sntran** [comment]
Hi there,

I switched to use 4.3.x recently and was surprised of how slow it was compared to 4.2.1, using the exact example in README.

### 4.2.1

```js
import { hash, verify } from "https://deno.land/x/scrypt@v4.2.1/mod.ts";

console.time("hash");
const hashResult = hash("password");
console.timeEnd("hash");

console.time("verify");
const verifyResult = verify("password", hashResult);
console.timeEnd("verify");
```

```shell
$ deno run -A scrypt.js
hash: 221ms
verify: 137ms
```

### 4.3.4

```js
import { hash, verify } from "https://deno.land/x/scrypt@v4.3.4/mod.ts";

console.time("hash");
const hashResult = hash("password");
console.timeEnd("hash");

console.time("verify");
const verifyResult = verify("password", hashResult);
console.timeEnd("verify");
```

```shell
$ deno run -A scrypt.js
hash: 1684ms
verify: 1199ms
```

I'm on recent Mac OSX.

**Comment by oplik0** [comment]
(I responded in #14 instead)
